internal/validations/clusterlogforwarder/validate_clusterlogforwarder_service_account.go

```diff
@@ -67,7 +67,10 @@ func getServiceAccount(name, namespace string, k8sClient client.Client) (*corev1
 // ValidateServiceAccountPermissions validates a service account for permissions to collect
 // inputs specified by the CLF.
 // ie. collect-application-logs, collect-audit-logs, collect-infrastructure-logs
-func validateServiceAccountPermissions(k8sClient client.Client, inputs sets.String, serviceAccount *corev1.ServiceAccount, clfNamespace, name string) error {
+func validateServiceAccountPermissions(k8sClient client.Client, inputs sets.String, noOfReceiverInputs int, serviceAccount *corev1.ServiceAccount, clfNamespace, name string) error {
+	if inputs.Len() == 0 && noOfReceiverInputs > 0 {
```

Comment:
`if inputs.Len() == 0 && noOfReceiverInputs > 0` can be evaluated without entering this function

Reply:
I realized that as well but chose this implementation to defer the evaluation to one place instead of conditionals outside of the function and within.  I'm not hard pressed to leave it one way or the other if you think its better to not get this far

Reply:
What if we have inputs in addition to receiver inputs?  Doesn't the way it is now allow consolidation of validation in one place?